lldb/source/Target/ThreadPlanCallFunction.cpp

```diff
@@ -496,8 +496,8 @@ bool ThreadPlanCallFunction::BreakpointsExplainStop() {
 #ifdef LLDB_ENABLE_SWIFT
       ConstString persistent_variable_name(
           persistent_state->GetNextPersistentVariableName(/*is_error*/ true));
-      if (m_return_valobj_sp = SwiftLanguageRuntime::CalculateErrorValue(
-              frame_sp, persistent_variable_name)) {
+      if ((m_return_valobj_sp = SwiftLanguageRuntime::CalculateErrorValue(
+               frame_sp, persistent_variable_name))) {
```

Comment:
Another case of assignment inside an `if`. This one looks like it should be assignment, not equality.

Reply:
Is the problem here that it's not a declaration? I.e., would `if (auto m_return_valobj_sp = SwiftLa...)` trigger a warning?

Reply:
Correct, a declaration would not trigger the warning. I don't know the history, but it seems clang uses additional parenthesis as a way of saying "this assignment inside an `if` is intentional".